### PR TITLE
🐞 Fix comet HPO

### LINKED ISSUE #593: HPO with Comet backend does not work
**Describe the bug**

- HPO fails with key error when used with comet.

**To Reproduce**

```bash
python tools/hpo/sweep.py --model stfpm --sweep_config tools/hpo/configs/comet.yaml 
```

**Stack Trace**
```bash

COMET INFO: COMET_OPTIMIZER_ID=ce90bba8fae04d5fa9875ee119b080ed
COMET INFO: Using optimizer config: {'algorithm': 'bayes', 'configSpaceSize': 4, 'endTime': None, 'id': 'ce90bba8fae04d5fa9875ee119b080ed', 'lastUpdateTime': None, 'maxCombo': 20, 'name': 'ce90bba8fae04d5fa9875ee119b080ed', 'parameters': {'dataset.image_size': {'type': 'discrete', 'values': [128, 256]}, 'model.backbone': {'type': 'categorical', 'values': ['resnet18', 'wide_resnet50_2']}}, 'predictor': None, 'spec': {'gridSize': 10, 'maxCombo': 20, 'metric': 'pixel_AUROC', 'minSampleSize': 100, 'objective': 'maximize', 'retryAssignLimit': 0, 'retryLimit': 1000}, 'startTime': 7711165905, 'state': {'mode': None, 'seed': None, 'sequence': [], 'sequence_i': 0, 'sequence_pid': None, 'sequence_retry': 0, 'sequence_retry_count': 0}, 'status': 'running', 'suggestion_count': 0, 'trials': 1, 'version': '2.0.1'}
COMET WARNING: Comet has disabled auto-logging functionality as it has been imported after the following ML modules: tensorboard, torch. Metrics and hyperparameters can still be logged using comet_ml.log_metrics() and comet_ml.log_parameters()
COMET ERROR: Failed to calculate active processors count. Fall back to default CPU count 1
COMET INFO: Experiment is live on comet.com https://www.comet.com/ashwin-vaidya/stfpm-hazelnut/194c327a17854bcaa634fb971144a26b

CometLogger will be initialized in online mode
Traceback (most recent call last):
  File "./tools/hpo/sweep.py", line 144, in <module>
    sweep.run()
  File "./tools/hpo/sweep.py", line 106, in run
    set_in_nested_config(self.config, param.split("."), run_params[param])
  File "/usr/local/lib/python3.7/dist-packages/anomalib/utils/sweep/config.py", line 133, in set_in_nested_config
    get_from_nested_config(config, keymap[:-1])[keymap[-1]] = value
TypeError: 'str' object does not support item assignment
COMET INFO: Optimizer metrics is 'pixel_AUROC' but no logged values found. Experiment ignored in sweep.
COMET INFO: ---------------------------
COMET INFO: Comet.ml Experiment Summary
COMET INFO: ---------------------------
COMET INFO:   Data:
COMET INFO:     display_summary_level : 1
COMET INFO:     url                   : https://www.comet.com/ashwin-vaidya/stfpm-hazelnut/194c327a17854bcaa634fb971144a26b
COMET INFO:   Others:
COMET INFO:     optimizer_count        : 1
COMET INFO:     optimizer_id           : ce90bba8fae04d5fa9875ee119b080ed
COMET INFO:     optimizer_metric       : pixel_AUROC
COMET INFO:     optimizer_metric_value : 1
COMET INFO:     optimizer_name         : ce90bba8fae04d5fa9875ee119b080ed
COMET INFO:     optimizer_objective    : minimum
COMET INFO:     optimizer_parameters   : {"dataset.image_size": 256, "model.backbone": "resnet18"}
COMET INFO:     optimizer_pid          : b0a87dc3b62fb6abe9ac181c7b00bdd69a1adc44
COMET INFO:     optimizer_process      : 981
COMET INFO:     optimizer_trial        : 1
COMET INFO:     optimizer_version      : 2.0.1
COMET INFO:   Parameters:
COMET INFO:     dataset.image_size : 256
COMET INFO:     model.backbone     : resnet18
COMET INFO:   Uploads:
COMET INFO:     environment details      : 1
COMET INFO:     filename                 : 1
COMET INFO:     git metadata             : 1
COMET INFO:     git-patch (uncompressed) : 1 (253 bytes)
COMET INFO:     installed packages       : 1
COMET INFO:     os packages              : 1
COMET INFO:     source_code              : 1 (5.35 KB)
COMET INFO: ---------------------------
COMET WARNING: Comet has disabled auto-logging functionality as it has been imported after the following ML modules: tensorboard, torch. Metrics and hyperparameters can still be logged using comet_ml.log_metrics() and comet_ml.log_parameters()
COMET INFO: Uploading metrics, params, and assets to Comet before program termination (may take several seconds)
COMET INFO: The Python SDK has 3600 seconds to finish before aborting...
```

- OS: Ubuntu 20.1

**Possible Solution**

1. The replace config method in sweep might be `wandb` specific. We will have to modify it to match `comet` keys.

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Description

- Fix comet HPO
- Minor refactor + pass entity/workspace from CLI
- Fix metrics callback in benchmarking

- Fixes #593 

## Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes

### CODE REVIEW

File: tools/hpo/utils/backends.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
+"""Sweep Backends."""
```
Comment (samet-akcay) [inline tools/hpo/utils/backends.py:1]: Can `WandbSweep` and `CometSweep` be considered as entrypoint? If not, can these be added to `anomalib/utils/sweep`?

Comment (samet-akcay) [reply tools/hpo/utils/backends.py:1]: Not sure. @djdameln any thoughts?

Comment (djdameln) [reply tools/hpo/utils/backends.py:1]: I would suggest either `anomalib/utils/hpo/backends.py` or `anomalib/utils/hpo/runners.py`

**Review by djdameln — APPROVED** [review]
I'm fine with merging this once we settle on the naming

File: tools/hpo/utils/backends.py (hunk 1 above)
Comment (samet-akcay) [reply tools/hpo/utils/backends.py:1]: As long as they are placed into `anomalib/utils/hpo/` or `anomalib/utils/sweep`, I don't mind what the modules are called.

**Review by djdameln — APPROVED** [review]
Thanks!

File: tools/hpo/utils/backends.py (hunk 1 above)
Comment (ashwinvaidya17) [reply tools/hpo/utils/backends.py]: @samet-akcay I have moved it to `anomalib/utils/hpo/runners.py`

Comment (samet-akcay) [reply tools/hpo/utils/backends.py]: Thanks

**Review by samet-akcay — APPROVED** [review]

**Pull request merged**